Kegledger and kegmanager apps, round 1

yourteamneedsadam opened the pull request:
@paulcollinsiii Turns out I've forgotten a lot of Django and I don't have my other laptop with me to use the reference repo from our private github, so this will probably need some reworking to get it to match your standard practices. The test suite is mostly empty pending review of the model design and test package structure. Ping me on the bits that need changes and I'll rebase them in, then finish up the tests.

paulcollinsiii commented on backend/kegstarter/kegledger/models.py:
```diff
@@ -0,0 +1,46 @@
+
+class LedgerEntry(models.Model):
+
+    amount = models.DecimalField(max_digits=5, decimal_places=2)
```
I'd actually max the max digits 8, because that allows for up to 10k of dollar tracking which should be plenty more than we're going to use.

yourteamneedsadam commented:
Fixed. I also added a few more tests and tweaked the test suite setup. There are a few uncovered lines but they're in __str__ methods and calls to Super in save methods that aren't being hit because there's not much functionality to test. I think that once we actually do something with the models those lines will get covered by the tests of that code and it seemed to make more sense to wait for that than to write tests that do stuff like call __str__() on all the factories just to get coverage.

paulcollinsiii commented:
Cool one I'm home I'll take a look at the rest of the code. I'll probably
work on getting some automated testing going his Travis too. Thanks!
On Feb 21, 2015 3:12 PM, "Adam Burns" <notifications@github.com> wrote:

> Fixed. I also added a few more tests and tweaked the test suite setup.
> There are a few uncovered lines but they're in *str* methods and calls to
> Super in save methods that aren't being hit because there's not much
> functionality to test. I think that once we actually do something with the
> models those lines will get covered by the tests of that code and it seemed
> to make more sense to wait for that than to write tests that do stuff like
> call *str*() on all the factories just to get coverage.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paulcollinsiii/django-kegstarter/pull/1#issuecomment-75398545>
> .
>

paulcollinsiii commented on backend/kegstarter/kegledger/models.py:
```diff
@@ -0,0 +1,46 @@
+            raise ValidationError('Only enter a guest name for users who are not registered.')
+
+    def save(self, *args, **kwargs):
+        self.clean()
```
FYI (I'll take care of updating these) `self.full_clean()`

yourteamneedsadam replied:
Ah, that makes sense. Implement custom validation in clean, but call full
clean on save so everything gets run. Keeps consistent behavior

On Sunday, February 22, 2015, Paul Collins <notifications@github.com> wrote:

> In backend/kegstarter/kegledger/models.py
> <https://github.com/paulcollinsiii/django-kegstarter/pull/1#discussion_r25135165>
> :
>
> > +    time = models.DateTimeField()
> > +    ledger = models.ForeignKey(Ledger)
> > +    notes = models.CharField(max_length=600, blank=True)
> > +    # No barriers to beer; you don't need an account to contribute to the keg fund.
> > +    user = models.ForeignKey(User, blank=True, null=True)
> > +    # Without an account we need your name so we can find your contributions later.
> > +    guest_name = models.CharField(max_length=200, blank=True)
> > +
> > +    def clean(self):
> > +        if self.user is None and not self.guest_name:
> > +            raise ValidationError('Either select a registered user or enter a guest name.')
> > +        if self.guest_name and self.user:
> > +            raise ValidationError('Only enter a guest name for users who are not registered.')
> > +
> > +    def save(self, *args, **kwargs):
> > +        self.clean()
>
> FYI (I'll take care of updating these) self.full_clean()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paulcollinsiii/django-kegstarter/pull/1/files#r25135165>
> .
>

paulcollinsiii commented on backend/kegstarter/kegmanager/models.py:
```diff
@@ -0,0 +1,86 @@
+        return '{}: {} gallons of {}'.format(self.purchase_date, self.gallons, self.beverage_name)
+
+
+class Poll(models.Model):
```
Poll and Ratings I think are also contrib apps. I need to check but they should be in their own locations.

yourteamneedsadam replied:
I checked contrib and didn't see them. Are you thinking something like
their own app plus gfks to make them work with arbitrary models, or do you
just mean they should be in a different models file?

On Sunday, February 22, 2015, Paul Collins <notifications@github.com> wrote:

> In backend/kegstarter/kegmanager/models.py
> <https://github.com/paulcollinsiii/django-kegstarter/pull/1#discussion_r25135248>
> :
>
> > +    """
> > +    beverage_name = models.CharField(max_length=200)
> > +    # django-measurements doesn't support Django 1.7, or we'd use that to make a MeasurementField. We force a single
> > +    # unit of measurement to make it easier to write a migration when django-measurements gets updated.
> > +    gallons = models.DecimalField(max_digits=4, decimal_places=2)
> > +    price = models.DecimalField(max_digits=5, decimal_places=2)
> > +    purchase_date = models.DateField(blank=True, null=True)  # Kegs in active polls won't have been purchased yet.
> > +    ratings = models.ManyToManyField(User, through='Rating')
> > +    tap = models.ForeignKey(Tap)
> > +    vendor_name = models.CharField(max_length=200)
> > +
> > +    def __str__(self):
> > +        return '{}: {} gallons of {}'.format(self.purchase_date, self.gallons, self.beverage_name)
> > +
> > +
> > +class Poll(models.Model):
>
> Poll and Ratings I think are also contrib apps. I need to check but they
> should be in their own locations.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paulcollinsiii/django-kegstarter/pull/1/files#r25135248>
> .
>

paulcollinsiii commented:
@yourteamneedsadam oh my bad, that's right django did finally remove those from contrib since they don't really belong in django itself. Anyhow I'm on my own branch moving some of the stuff around right now. Gimme a sec to push it up so you can see what I'm working on now (beware I'll be rebasing and force pushing that branch up over the course of today)

yourteamneedsadam commented:
Cool, sounds good

On Sunday, February 22, 2015, Paul Collins <notifications@github.com> wrote:

> @yourteamneedsadam <https://github.com/yourteamneedsadam> oh my bad,
> that's right django did finally remove those from contrib since they don't
> really belong in django itself. Anyhow I'm on my own branch moving some of
> the stuff around right now. Gimme a sec to push it up so you can see what
> I'm working on now (beware I'll be rebasing and force pushing that branch
> up over the course of today)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paulcollinsiii/django-kegstarter/pull/1#issuecomment-75454326>
> .
>

paulcollinsiii commented:
Pulled into #2
